### PR TITLE
Expose epsilon for cost gradient finite difference approximation

### PR DESCRIPTION
**Pull request by Levi-Armstrong**
This fixes issue #70

### CODE REVIEW

**Comment by Levi-Armstrong** [comment]
@awinkler what are your thoughts on this PR?

File: ifopt_core/src/problem.cc [hunk 1]
```diff
@@ -97,16 +97,13 @@ Problem::EvaluateCostFunction (const double* x)
 }
 
 Problem::VectorXd
-Problem::EvaluateCostFunctionGradient (const double* x, bool use_finite_difference_approximation)
+Problem::EvaluateCostFunctionGradient (const double* x, bool use_finite_difference_approximation, double epsilon)
```
Comment (awinkler) [inline ifopt_core/src/problem.cc:100]: can you add a default value also to this epsilon, so the interface remains backwards compatible? Then feel free to merge this PR.

Comment (Levi-Armstrong) [reply ifopt_core/src/problem.cc:100]: I may be missing something, but it has a default value defined in the .h file. Is that what you are wanting?

Comment (awinkler) [reply ifopt_core/src/problem.cc:100]: Ah, that's it, missed it. Yeah, feel free to merge and if you need to update distros. Thanks so much!🙏

Comment (Levi-Armstrong) [reply ifopt_core/src/problem.cc:100]: I have release the latest on both noetic and foxy.

**Pull request merged**